### PR TITLE
#[UIC-1594]:fixed modal slice loading issue

### PR DESCRIPTION
**Pull request by jitendra-kumawat**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Fixed modal slice loading issue

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
NA

### TEST PLAN
Try to select different  item on publishers so that when subscribe modal  slice open  then it  should show loading state.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@ankursinghal2005

### CODE REVIEW

File: superset/assets/src/dashboard/components/Dashboard.jsx [hunk 1]
```diff
@@ -247,14 +247,17 @@ class Dashboard extends React.PureComponent {
...
 
-  updateModalProps(slice_id, showModal,status = undefined) {
+  updateModalProps(slice_id, showModal, status = undefined) {
     this.showModal = showModal
     this.modalChart = this.props.charts[slice_id];
     if (this.modalChart) {
+      if (status) {
+        this.modalChart.chartStatus = status
+      }
       this.modalChartStatus = status ? status : this.modalChart.chartStatus;
```
Comment (ankursinghal2005) [inline superset/assets/src/dashboard/components/Dashboard.jsx]: We can remove variable modalChartStatus and use directly this.modalChart.chartStatus or directly assign this.modalChart.chartStatus value in this.modalChartStatus variable.

File: superset/assets/src/dashboard/components/Dashboard.jsx [hunk 2]
```diff
@@ -247,14 +247,17 @@ class Dashboard extends React.PureComponent {
...
     }
   }
 
-  updateModalProps(slice_id, showModal,status = undefined) {
+  updateModalProps(slice_id, showModal, status = undefined) {
     this.showModal = showModal
     this.modalChart = this.props.charts[slice_id];
     if (this.modalChart) {
+      if (status) {
+        this.modalChart.chartStatus = status
```
Comment (ankursinghal2005) [inline superset/assets/src/dashboard/components/Dashboard.jsx]: we can add semi-column also...

File: superset/assets/src/dashboard/components/Dashboard.jsx (hunk 1 above)
Comment (jitendra-kumawat) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: done

File: superset/assets/src/dashboard/components/Dashboard.jsx (hunk 2 above)
Comment (jitendra-kumawat) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: done

**Pull request merged**